Sources/ProcedureKit/ProcedureQueue.swift

```diff
@@ -137,129 +184,273 @@ open class ProcedureQueue: OperationQueue {
+        promise.complete()
+    }
 
-                        // Indirect dependencies are executed after
-                        // any previous mutually exclusive operation(s)
-                        $0.add(dependencies: previousMutuallyExclusiveOperations)
+    /// Processes a Procedure's Conditions, and returns a Procedure.EvaluateConditions Procedure
+    /// (if there are Conditions present).
+    ///
+    /// - Parameter procedure: a Procedure that is being added to the ProcedureQueue
+    /// - Returns: the EvaluateConditions Procedure for the input Procedure
+    private func processConditions(procedure: Procedure) -> Procedure? {
+        guard procedure.conditions.count > 0 else { return nil }
 
-                        // Indirect dependencies are executed after
-                        // all direct dependencies
-                        $0.add(dependencies: procedure.directDependencies)
+        /// Check for mutual exclusion conditions
+        let manager = ExclusivityManager.sharedInstance
 
-                        // Only evaluate conditions after all indirect
-                        // dependencies have finished
-                        evaluator.addDependency($0)
-                    }
+        let mutuallyExclusiveConditions = procedure.conditions.filter { $0.isMutuallyExclusive }
+        var previousMutuallyExclusiveOperations = Set<Operation>()
 
-                    // Add indirect dependencies
-                    add(operations: indirectDependenciesToProcess)
-                }
+        for condition in mutuallyExclusiveConditions {
+            let category = "\(condition.category)"
+            if let previous = manager.add(procedure: procedure, category: category) {
+                previousMutuallyExclusiveOperations.insert(previous)
             }
-
-            // Add the evaluator
-            addOperation(evaluator)
         }
 
-        // Indicate to the operation that it is to be enqueued
-        procedure.willEnqueue(on: self)
+        // Create the condition evaluator
+        let evaluator = procedure.evaluateConditions()
 
-        delegate?.procedureQueue(self, willAddOperation: procedure)
-    }
-    // swiftlint:enable cyclomatic_complexity
-    // swiftlint:enable function_body_length
+        // Get the condition dependencies
+        let indirectDependencies = procedure.indirectDependencies
 
+        // If there are dependencies
+        if indirectDependencies.count > 0 {
 
-    /**
-     Adds the operations to the queue.
+            // Filter out the indirect dependencies which have already been added to the queue
+            let indirectDependenciesToProcess = indirectDependencies.filter { !self.operations.contains($0) }
 
-     - parameter ops: an array of `NSOperation` instances.
-     - parameter wait: a Bool flag which is ignored.
+            // Check to see if there are any which need processing
+            if indirectDependenciesToProcess.count > 0 {
```

Comment:
Yup. Carry-over from when this wasn't broken out into its own function (i.e. prior commits).